wlog/auditlog/audit3log/params.go

```diff
@@ -0,0 +1,220 @@
+		logger.StringValue(ProductVersionKey, productVersion)
+	})
+}
+
+func Stack(stack string) Param {
+	return paramFunc(func(entry wlog.LogEntry) {
+		entry.OptionalStringValue(StackKey, stack)
+	})
+}
+
+func Service(service string) Param {
+	return paramFunc(func(entry wlog.LogEntry) {
+		entry.OptionalStringValue(ServiceKey, service)
+	})
+}
+
+func Environment(environment string) Param {
+	return paramFunc(func(entry wlog.LogEntry) {
+		entry.OptionalStringValue(EnvironmentKey, environment)
+	})
+}
+
+func ProducerType(producerType AuditProducerType) Param {
+	return paramFunc(func(entry wlog.LogEntry) {
+		entry.OptionalStringValue(ProducerTypeKey, string(producerType))
+	})
+}
+
+func Organizations(organizations ...AuditOrganizationType) Param {
+	return paramFunc(func(entry wlog.LogEntry) {
+		entry.ObjectValue(OrganizationsKey, organizations, reflect.TypeOf(organizations))
+	})
+}
+
+func EventID(eventID string) Param {
+	return paramFunc(func(entry wlog.LogEntry) {
+		entry.OptionalStringValue(EventIDKey, eventID)
+	})
+}
+
+func UserAgent(userAgent string) Param {
+	return paramFunc(func(entry wlog.LogEntry) {
+		entry.OptionalStringValue(UserAgentKey, userAgent)
+	})
+}
+
+func Categories(categories ...string) Param {
+	return paramFunc(func(entry wlog.LogEntry) {
+		entry.StringListValue(CategoriesKey, categories)
+	})
+}
+
+func Entities(entities ...interface{}) Param {
+	return paramFunc(func(entry wlog.LogEntry) {
+		entry.ObjectValue(EntitiesKey, entities, reflect.TypeOf(entities))
+	})
+}
+
+func Users(users ...AuditContextualizedUserType) Param {
+	return paramFunc(func(entry wlog.LogEntry) {
+		entry.ObjectValue(UsersKey, users, reflect.TypeOf(users))
+	})
+}
+
+func Origins(origins ...string) Param {
+	return paramFunc(func(entry wlog.LogEntry) {
+		entry.StringListValue(OriginsKey, origins)
+	})
+}
+
+func SourceOrigin(sourceOrigin string) Param {
+	return paramFunc(func(entry wlog.LogEntry) {
+		entry.OptionalStringValue(SourceOriginKey, sourceOrigin)
+	})
+}
+
+func UID(uid string) Param {
+	return paramFunc(func(entry wlog.LogEntry) {
+		entry.OptionalStringValue(wlog.UIDKey, uid)
+	})
+}
+
+func SID(sid string) Param {
+	return paramFunc(func(entry wlog.LogEntry) {
+		entry.OptionalStringValue(wlog.SIDKey, sid)
+	})
+}
+
+func TokenID(tokenID string) Param {
+	return paramFunc(func(entry wlog.LogEntry) {
+		entry.OptionalStringValue(wlog.TokenIDKey, tokenID)
+	})
+}
+
+func TraceID(traceID string) Param {
+	return paramFunc(func(entry wlog.LogEntry) {
+		entry.OptionalStringValue(wlog.TraceIDKey, traceID)
+	})
+}
+
+func OtherUIDs(otherUIDs ...string) Param {
+	return paramFunc(func(entry wlog.LogEntry) {
+		entry.StringListValue(OtherUIDsKey, otherUIDs)
+	})
+}
+
+func Origin(origin string) Param {
+	return paramFunc(func(entry wlog.LogEntry) {
+		entry.OptionalStringValue(OriginKey, origin)
+	})
+}
+
+func RequestParam(key string, levels []AuditSensitivityType, payload interface{}) Param {
```

Comment:
Are the levels required? I'm wondering if `RequestParam(key string, payload interface{}, levels ...AuditSensitivityType)` would be a friendlier syntax

Reply:
Levels are required yes but I can still change to use your syntax, will do this today